### PR TITLE
Merge Development

### PR DESCRIPTION
**Pull request by er-contreras**
- A login page.
- A registration page.
- A food list (with all CRUD implementation, except for 'update').
- A recipes list (with all CRUD implementation, except for 'update').
- A general shopping list view (all missing food for all your recipes and total price).
- A list of all public recipes from other users with their names and total prices.

### CODE REVIEW

**Review by mke2111 — CHANGES_REQUESTED** [review]
### Hey @eri8-9 & @tiagorahal, 

**This is a great project you have going on, well-done lads**. 

   
**However, there are a few sections to improve before your project is approved.** 🔴 
<hr>

## Required Changes ♻️
_Check the inline comments under the review for sections to improve._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 1]
```diff
@@ -1,24 +1,58 @@
...
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+# Recipe App
 
-Things you may want to cover:
+> This projects was built with ruby
 
-* Ruby version
+Additional description about the project and its features.
```
Comment (mke2111) [inline README.md:7]: - [ ] Add a clear project description to the readme file. Remember this is part of what constitutes a professional readme file.

File: app/views/foods/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<section id="food">
+  <div class="btn-container">
+    <% if user_signed_in? %>
+      <%= button_to 'Add Food', '/foods/new', method: :get %>
+    <% end %>
+  </div>
```
Comment (mke2111) [inline app/views/foods/index.html.erb:6]: - [ ] When a user adds new food details, on submitting the form, they are redirected to the same form after submission. They should be redirected to the foods list instead.
- [ ] Also, on submission of the form, the foods list does not get updated i.e. the new entry does not appear. Give this a fix.

File: app/views/public_recipes/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<section class="public-recipes-section">
+  <h2 class="pr-header">Public Recipes</h2>
+  <div class="recipe-card-container">
+    <% @recipes.each do |recipe| %>
+      <%= render 'public_recipe_card', recipe: recipe %>
+    <% end %>
+  </div>
```
Comment (mke2111) [inline app/views/public_recipes/index.html.erb:7]: - [ ] Unfortunately, your recipe index page does not match the given wireframe. [Here is the original wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/images/recipe_index.png).
Screenshot from 2021-12-17 04-30-15

File: app/views/recipes/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        <% end %>
+      </div>
+    <% end %>
+    </div>
+  </div>
+
+  <div class="btns-container">
+    <div class="m-4">
+      <%= button_to 'Generate shopping list', '/foods/new', method: :get %>
+    </div>
```
Comment (mke2111) [inline app/views/recipes/show.html.erb:29]: - [ ] On submission of this form, the foods list does not get updated i.e. the new entry does not appear in the general shopping list. Give this a fix.

File: app/views/recipes/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<section class="recipes-section">
+  <h2>Recipes</h2>
+  <div class="add-recipe-container">
+    <%= button_to 'Add Recipe', new_recipe_path, method: :get, class: 'add-recipe-btn' %>
+  </div>
```
Comment (mke2111) [inline app/views/recipes/index.html.erb:5]: - [ ] Unfortunately, your recipe index page does not match the given wireframe. [Here is the original wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/images/recipe_index.png).
Screenshot from 2021-12-17 04-30-15

File: config/routes.rb [hunk 6]
```diff
@@ -1,3 +1,14 @@
...
+  devise_for :users
+  resources :foods
+  get 'public_recipes', to: "public_recipes#index", as: "public_recipes"
+  get 'general_shopping_list', to: "general_shopping_list#index", as: "general_shopping_list"
+  resources :recipes do
+    resources :recipe_foods, except: [:show]
+  end 
+  resources :foods, except: [:update, :show]
+  get 'users/index'
+  get 'users/show'
```
Comment (mke2111) [inline config/routes.rb:12]: - [ ] You have added two routes for `:foods` serving the same purpose. Check lines 4 and 10. I suggest you keep line 10 and delete line 4.

**Review by oloomoses — APPROVED** [review]
Hi @eri8-9, 

Well done on making all the suggested changes.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

_Slack: Oloo Moses_
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**